src/shared/lib/extendCBioPortalAPI.ts

```diff
@@ -0,0 +1,113 @@
+import { SiteError } from 'AppStore';
+import { getBrowserWindow } from 'cbioportal-frontend-commons';
+import { Simulate } from 'react-dom/test-utils';
+import error = Simulate.error;
+import autobind from 'autobind-decorator';
+import request from 'superagent';
+
+type Constructor<T> = new (...args: any[]) => T;
+
+export function extendCBioPortalAPI<T extends Constructor<{}>>(Base: T) {
+    return class extends Base {
+        defaultError: (error: any) => Partial<SiteError>;
+
+        handleErrorsGlobally(errorConfig: Partial<SiteError> = {}) {
+            const self = this;
+
+            // global error handling is default
+            // only do this if they pass special configuration
+            if (errorConfig) {
+                // @ts-ignore (it's private, but we CAN access it)
```

Comment:
why keep it private then? maybe it should be protected?

Reply:
it's private as defined by SuperAgent's typing.  but in fact it can be accessed